docusaurus/docs/create-a-plugin/extend-a-plugin/instrument-a-backend-plugin.md

```diff
@@ -0,0 +1,421 @@
+
+:::
+
+### Inspect logs locally
+
+Logs from a backend plugin will be consumed by the connected Grafana instance and included in the Grafana server log.
+
+Each log message for a backend plugin will include a name, `logger=plugin.<plugin id>`, e.g.
+
+```shell
+DEBUG[11-14|15:26:26] Debug msg     logger=plugin.grafana-helloworld-app someID=1
+INFO [11-14|15:26:26] Info msg      logger=plugin.grafana-helloworld-app queryType=default
+WARN [11-14|15:26:26] Warning msg   logger=plugin.grafana-helloworld-app key=value
+ERROR[11-14|15:26:26] Error msg     logger=plugin.grafana-helloworld-app error=An error occurred
+```
+
+You can enable [debug logging in your Grafana instance](https://grafana.com/docs/grafana/latest/troubleshooting/#troubleshoot-with-logs) and that will normally output a huge amount of information and make it hard to find the logs related to a certain plugin. However, using a named logger makes it convenient to enable debug logging only for a certain named logger and/or plugin:
+
+```
+[log]
+filters = plugin.<plugin id>:debug
+```
+
+Please refer to [Configure Grafana](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#log) for more details setting up logging.
+
+Further, see [How to collect and visualize logs, metrics and traces](#collect-and-visualize-logs-metrics-and-traces).
+
+## Metrics
+
+Metrics are quantifiable measurements that reflect the health and performance of applications or infrastructure.
+
+Consider using metrics to provide real-time insight into the state of resources. If you want to know how responsive your plugin is or identify anomalies that could be early signs of a performance issue, metrics are a key source of visibility.
+
+### Metric types
+
+See [Prometheus metric types](https://prometheus.io/docs/concepts/metric_types/) for a list and description of the different metric types you can use and when to use them.
+
+There are many possible types of metrics that can be tracked. One popular method for defining metrics is the [RED method](https://grafana.com/blog/2018/08/02/the-red-method-how-to-instrument-your-services/).
+
+### Implement metrics in your plugin
+
+[Grafana plugin SDK for Go](../../introduction/grafana-plugin-sdk-for-go.md) uses the [Prometheus instrumentation library for Go applications](https://github.com/prometheus/client_golang). Any custom metric registered with the [default registry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables) will be picked up by the SDK and exposed through the [Collect metrics capability](../../introduction/backend.md#collect-metrics).
+
+For convenience, It's recommended to use the [promauto package](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promauto) when creating custom metrics since it will automatically register the metric in the [default registry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables).
```

Comment:
```suggestion
For convenience, It's recommended to use the [promauto package](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promauto) when creating custom metrics since it automatically registers the metric in the [default registry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables).
```

Reply:
```suggestion
For convenience, it's recommended to use the [promauto package](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promauto) when creating custom metrics since it automatically registers the metric in the [default registry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables) and expose them to Grafana.
```